bson/src/main/org/bson/codecs/pojo/ClassModelBuilder.java

```diff
@@ -321,6 +321,9 @@ private void validatePropertyModels(final String declaringClass, final List<Prop
         Map<String, Integer> propertyWriteNameMap = new HashMap<String, Integer>();
 
         for (PropertyModel<?> propertyModel : propertyModels) {
+            if (propertyModel.hasError()) {
+                throw new CodecConfigurationException(propertyModel.getError());
```

Comment:
We now throw the error during the final validation stage.

Reply:
I'm not clear on what's happening here.  The idea is that delaying validation can sometimes result in not reporting a spurious validation error.  But how does that actually happen?  Is the error cleared somehow?

Reply:
The issue is when we inspect a class and its properties we'd immediately fail if there were two conflicting types for a property.  This happened as soon as the code saw the error, regardless of any logic that conventions may apply. Its the conventions that can remove / modify propertyModelBuilders and their properties. In this case the user wanted the convention to totally ignore the field.

To work around the immediate throwing of an error, meant we'd have to keep the error message around on a `PropertyModelBuilder`.  That way when validation occurs during the final stage of creating the `ClassModel` if the invalid `PropertyModel` still is included the exception can be thrown.

I'm not 100% happy about the fix - because I think the real issue is the read and write types have to align in property models.  Also, it would potentially be nicer if a single overridden setter could be ignored, rather than the field as a whole.